book/src/requirements.md

```diff
@@ -0,0 +1,166 @@
+
+**Rationale - UX:** A separate shielded pool associated with a new address type makes it easier to reason about migration.
+
+### ProtoR4. Orchard Privacy Protections are Independent of Soundness.
+
+**Requirement:** Orchard's privacy protections are independent of proving system soundness (in the sense that the security arguments are disentangled from each other and can be separately verified) so that a soundness compromise does not compromise any privacy protections.
+
+**Rationale - UX:** Users can reasonably rely on the full privacy protections of Orchard even in the face of an Orchard soundness compromise, such as a successful counterfeiting attack.
+
+## Non-functional Requirements
+
+Non-functional requirements may not be verifiable from the protocol specification itself, but never-the-less define the goals, processes, or assessments used in delivering the Orchard protocol.
+
+### Technical Strategy Requirements
+
+These requirements define and rationalize some high-level goal which may not be apparent from the design specification.
+
+#### TSR1. Sapling-equivalent Functionality
+
+**Requirement:** Unless other product or engineering concerns dictate, the functionality of
+Orchard should match that of the Sapling shielded protocol.
+
+**Rationale - UX:** By being functional equivalent to Sapling, the existing Sapling
+user base can immediately carry over most conceptual knowledge and UX habits from
+Sapling to Orchard.
+
+**Rationale - Technical Strategy:** Orchard is the first step in removing the trusted setup
+attack surface as well as in deploying Halo 2. By targeting Sapling shielded protocol functionality,
+the scope of Orchard R&D is greatly simplified to lower the strategic technical risk.
+
+#### TSR2. Differences from Sapling Design Document
+
+**Requirement:** Every observable difference from Sapling from the perspective of a Mobile Wallet SDK, a zcashd RPC developer, or any user of any known services built with those components, must be documented in a "Differences from Sapling" document in this repository. Any change that impacts a product-level requirement must exist in this document as an explicit requirement.
+
+**Rationale - Technical Strategy:** Any deviation from Sapling needs to be clearly communicated to users and developers and reviewed by protocol engineers, product owners, and ecosystem stakeholders.
+
+#### TSR3. Halo 2 Deployment
+
+**Requirement:** Orchard must use Halo 2 for zero-knowledge proof validation.
+
+**Rationale - Technical Strategy:** By selecting Halo 2 as the ZKP platform for Orchard, the Orchard deployment provides a forcing function to productionize Halo 2, thus unlocking potential future applications of recursion/proof-carrying data while limiting short-term risk.
+
+#### TSR4. User Opt-in upon Activation
+
+**Requirement:** Users may individually shield funds or migrate into the Orchard Pool as soon as the feature activates.
```

Comment:
To me, "opt-in" (from the title of the issue) means "the user can decide to keep using the old thing, and has to explicitly decide to switch to the new thing, it's not forced on them" but I believe wallets may start auto-sheilding to Orchard. Does TSR4 mean wallets must make this opt-in, i.e. the wallet will keep working exactly as before until the user clicks a button to switch over to Orchard? I think some clarification is needed.

Reply:
Perhaps the title should be "Availability to users upon activation"